src/components/Tooltip/index.js

```diff
@@ -1,121 +1,114 @@
 import TooltipSense from './TooltipSense';
 import * as DeviceCapabilities from '../../libs/DeviceCapabilities';
-import compose from '../../libs/compose';
-import withLocalize from '../withLocalize';
+import usePrevious from '../../hooks/usePrevious';
+import useLocalize from '../../hooks/useLocalize';
+import useWindowDimensions from '../../hooks/useWindowDimensions';
+
+const hasHoverSupport = DeviceCapabilities.hasHoverSupport();
 
 /**
  * A component used to wrap an element intended for displaying a tooltip. The term "tooltip's target" refers to the
  * wrapped element, which, upon hover, triggers the tooltip to be shown.
+ * @param {propTypes} props
+ * @returns {ReactNodeLike}
  */
-class Tooltip extends PureComponent {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            // Is tooltip already rendered on the page's body? This happens once.
-            isRendered: false,
-
-            // Is the tooltip currently visible?
-            isVisible: false,
-
-            // The distance between the left side of the wrapper view and the left side of the window
-            xOffset: 0,
-
-            // The distance between the top of the wrapper view and the top of the window
-            yOffset: 0,
-
-            // The width and height of the wrapper view
-            wrapperWidth: 0,
-            wrapperHeight: 0,
-        };
-
-        // Whether the tooltip is first tooltip to activate the TooltipSense
-        this.isTooltipSenseInitiator = false;
-        this.animation = new Animated.Value(0);
-        this.hasHoverSupport = DeviceCapabilities.hasHoverSupport();
-
-        this.showTooltip = this.showTooltip.bind(this);
-        this.hideTooltip = this.hideTooltip.bind(this);
-        this.updateBounds = this.updateBounds.bind(this);
-        this.isAnimationCanceled = React.createRef(false);
-    }
-
-    // eslint-disable-next-line rulesdir/prefer-early-return
-    componentDidUpdate(prevProps) {
-        // if the tooltip text changed before the initial animation was finished, then the tooltip won't be shown
-        // we need to show the tooltip again
-        if (this.state.isVisible && this.isAnimationCanceled.current && this.props.text && prevProps.text !== this.props.text) {
-            this.isAnimationCanceled.current = false;
-            this.showTooltip();
-        }
-    }
-
-    /**
-     * Update the tooltip bounding rectangle
-     *
-     * @param {Object} bounds - updated bounds
-     */
-    updateBounds(bounds) {
-        if (bounds.width === 0) {
-            this.setState({isRendered: false});
-        }
-        this.setState({
-            wrapperWidth: bounds.width,
-            wrapperHeight: bounds.height,
-            xOffset: bounds.x,
-            yOffset: bounds.y,
-        });
-    }
+function Tooltip(props) {
+    const {children, numberOfLines, maxWidth, text, renderTooltipContent, renderTooltipContentKey} = props;
```

Comment:
I used this destructuring syntax, because the props object is used below with `result` from lodash:

```
shiftHorizontal={_.result(props, 'shiftHorizontal')}
shiftVertical={_.result(props, 'shiftVertical')}
```